core/src/main/java/hudson/security/AuthenticationProcessingFilter2.java

```diff
@@ -103,7 +103,12 @@ protected void onUnsuccessfulAuthentication(HttpServletRequest request, HttpServ
         LOGGER.log(Level.FINE, "Login attempt failed", failed);
         Authentication auth = failed.getAuthentication();
         if (auth != null) {
-            SecurityListener.fireFailedToLogIn(auth.getName());
+        	//is client behind something?
+			String ipAddress = request.getHeader("X-FORWARDED-FOR");  
```

Comment:
This should be taken care of by the server. We don't use this for the `Cause.RemoteCause` either, for example:
https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/jenkins/model/ParameterizedJobMixIn.java#L258

See https://github.com/jenkinsci/winstone/pull/13